knot-thing/src/main/java/com/cesar/knot_sdk/KNoTAMQPFactory.kt

```diff
@@ -0,0 +1,69 @@
+package com.cesar.knot_sdk
+
+import org.jetbrains.anko.doAsync
+
+/**
+ * The creation of the KNoTAMQP class depends on a series of complicated
+ * operations, some of which involve network operations. This class is
+ * responsible solely to ease the creation of KNoTAMQP instances.
+ */
+class KNoTAMQPFactory() {
+
+    /**
+     * Configures the KNoT-AMQP instance.
+     * @param kNoTAMQP a non-configured instance of the KNoTAMQP class.
+     * @param callback a callback to return after the operations are done.
+     */
+    fun getKNoTProtocolMessager(
```

Comment:
I'd change this name to `create()` or `getKNoTAMQP()`.

Despite of being JavaScript, [here](https://github.com/CESARBR/knot-fog-connector/blob/master/src/network/AMQPConnectionFactory.js#L8) is one example of using `create()` in a factory and [here](https://github.com/CESARBR/knot-fog-connector/blob/master/src/index.js#L22) is one example of accessing it.